azure-pipelines.yml

```diff
@@ -21,6 +13,32 @@ jobs:
   steps:
   - template: azure-pipelines-steps.yml
 
+  - script: npm pack
+    displayName: 'npm pack'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+
+  - task: CopyFiles@2
+    inputs:
+      sourceFolder: '$(Build.SourcesDirectory)'
+      contents: '*.tgz' 
+      targetFolder: $(Build.ArtifactStagingDirectory)
+    displayName: 'Copy npm package to staging'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+
+  - task: CopyFiles@2
```

Comment:
Nit - you can combine the copyfiles steps - contents takes a multiline string which can be:
```
*.tgz
package.json
```